brainbeats-backend/Controllers/PlaylistsController.cs

```diff
@@ -0,0 +1,202 @@
+      } catch (Exception e) {
+        return Unauthorized($"Unauthenticated error: {e}");
+      }
+
+      try {
+        var result = await SearchPlaylistVertexQuery(name, email);
+        return Ok(result);
+      } catch (ArgumentException e) {
+        return BadRequest($"Malformed request: {e}");
+      } catch (ResponseException e) {
+        return BadRequest($"Error with Gremlin Query: {e}");
+      } catch (Exception e) {
+        return BadRequest($"Something went wrong: {e}");
+      }
+    }
+
+    // POST api/v2/playlists/create/{email}?seed=seed
+    [HttpPost("create/{email}")]
+    public async Task<IActionResult> CreatePlaylist(string email, string seed, [FromForm] PlaylistVertex p) {
+      try {
+        HttpContext.Request.Headers.TryGetValue("Authorization", out StringValues authorizationToken);
+        AuthConnection.Instance.ValidateToken(authorizationToken);
+      } catch (ArgumentException e) {
+        return BadRequest($"Malformed or missing authorization token: {e}");
+      } catch (Exception e) {
+        return Unauthorized($"Unauthenticated error: {e}");
+      }
+
+      try {
+        var result = await CreatePlaylistVertexQuery(email, p, seed);
+        return Ok(result);
+      } catch (ArgumentException e) {
+        return BadRequest($"Malformed request: {e}");
+      } catch (ResponseException e) {
+        return BadRequest($"Error with Gremlin Query: {e}");
+      } catch (Exception e) {
+        return BadRequest($"Something went wrong: {e}");
+      }
+    }
+
+    // PUT api/v2/playlists/add_beats/{playlistId}
+    [HttpPut("add_beats/{playlistId}")]
+    public async Task<IActionResult> UpdatePlaylistAddBeats(string playlistId, dynamic req) {
+      try {
+        HttpContext.Request.Headers.TryGetValue("Authorization", out StringValues authorizationToken);
+        AuthConnection.Instance.ValidateToken(authorizationToken);
+      } catch (ArgumentException e) {
+        return BadRequest($"Malformed or missing authorization token: {e}");
+      } catch (Exception e) {
+        return Unauthorized($"Unauthenticated error: {e}");
+      }
+
+      try {
+        JObject body = DeserializeRequest(req);
+
+        if (!body.ContainsKey("beatId")) {
+          return BadRequest("Request body must contain 'beatId'");
+        }
+
+        var result = await UpdatePlaylistAddBeat(playlistId, body.GetValue("beatId").ToString());
+        return Ok(result);
+      } catch (ArgumentException e) {
+        return BadRequest($"Malformed request: {e}");
+      } catch (ResponseException e) {
+        return BadRequest($"Error with Gremlin Query: {e}");
+      } catch (Exception e) {
+        return BadRequest($"Something went wrong: {e}");
+      }
+    }
+
+    // PUT api/v2/playlists/remove_beats/{playlistId}
+    [HttpPut("remove_beats/{playlistId}")]
+    public async Task<IActionResult> UpdatePlaylistRemoveBeats(string playlistId, dynamic req) {
```

Comment:
Route seems ambiguous. What about a 'remove_beat_from_playlist' with the details being put in a form. I could be in the wrong for this one.